core/src/types.ts

```diff
@@ -664,6 +672,36 @@ export type ClanGetForAdminResponse = Clan & {
   brutes: Pick<Brute, 'id' | 'name'>[],
 };
 
+// Clan Roles Types
+export type ClanRolePermissions = {
+  canAcceptJoinRequests: boolean;
+  canRejectJoinRequests: boolean;
+  canRemoveMembers: boolean;
+  canSelectWarFighters: boolean;
+  canPinThreads: boolean;
+  canUnpinThreads: boolean;
+  canDeletePosts: boolean;
+  canDeleteThreads: boolean;
+  canCreateRoles: boolean;
+  canChangeRoles: boolean;
+};
+
+export type ClanRoleWithMemberCount = ClanRole & {
+  memberCount: number;
+};
+
+export type ClanGetRolesResponse = ClanRoleWithMemberCount[];
+
+export type ClanCreateRoleRequest = Pick<ClanRole, 'name'> & Partial<ClanRolePermissions>;
+export type ClanCreateRoleResponse = ClanRole;
+
+export type ClanUpdateRoleRequest = Partial<Pick<ClanRole, 'name'>> & Partial<ClanRolePermissions>;
+export type ClanUpdateRoleResponse = ClanRole;
+
+export type ClanAssignRoleResponse = { success: boolean };
+export type ClanRemoveRoleResponse = { success: boolean };
```

Comment:
The response types should be exact with the handlers return types.
here you use the base types which include all direct fields.
To help, you should add return types to your server handlers, they will tell you if you add too much/too little fields